airbyte-cdk/python/airbyte_cdk/sources/streams/checkpoint/checkpoint_reader.py

```diff
@@ -111,19 +111,34 @@ def next(self) -> Optional[Mapping[str, Any]]:
         """
 
         try:
-            if self._current_slice is None:
-                self._current_slice = self._get_next_slice()
-                state_for_slice = self._cursor.select_state(self._current_slice)
-                if state_for_slice == {"__ab_full_refresh_sync_complete": True}:
-                    return None
-                else:
-                    return self._current_slice
             if self._read_state_from_cursor:
-                state_for_slice = self._cursor.select_state(self._current_slice)
-                if state_for_slice == {"__ab_full_refresh_sync_complete": True}:
-                    self._current_slice = self._get_next_slice()
+                if self._current_slice is None:
+                    next_slice = self._get_next_slice()
+                    state_for_slice = self._cursor.select_state(self._current_slice)
+                    if state_for_slice == {"__ab_full_refresh_sync_complete": True}:
+                        next_candidate_slice = None
+                        has_more = True
+                        while has_more:
+                            next_candidate_slice = self._get_next_slice()
+                            state_for_slice = self._cursor.select_state(next_candidate_slice)
+                            has_more = state_for_slice == {"__ab_full_refresh_sync_complete": True}
```

Comment:
There are many places where this part: `{"__ab_full_refresh_sync_complete": True}` of code could be reused, can we have the `constant` assigned to this value and then reused instead of duplicating the code?

like: 

```python
FULL_REFRESH_SYNC_COMPLETE: Dict = {"__ab_full_refresh_sync_complete": True}
```

Reply:
ah yes. that was the same comment on this [PR](https://github.com/airbytehq/airbyte/pull/39450) from Alex. I probably forked this PR out before I made that change to the constants, but will fix